dpc/src/circuits/outer_circuit_gadget.rs

```diff
@@ -317,7 +342,7 @@ pub fn execute_outer_circuit<C: Parameters, CS: ConstraintSystem<C::OuterScalarF
         )?;
 
         let birth_program_vk_field_elements =
-            birth_program_vk.to_constraint_field(cs.ns(|| "birth_death_program_vk_field_elements"))?;
+            birth_program_vk.to_constraint_field(cs.ns(|| "birth_birth_program_vk_field_elements"))?;
```

Comment:
Good catch. I feel that maybe our naming of variables in this loop is suboptimal. 

Is it true that in this part of the code, actually all `birth_program_vk_field_elements` is supposed to be something like `birth_or_death_program_vk_field_elements`?  Since the loop is over all the records:

```
for (j, input) in program_proofs
        .iter()
        .skip(C::NUM_INPUT_RECORDS)
        .enumerate()
        .take(C::NUM_OUTPUT_RECORDS)
```

should we change the names to make this clear?

Reply:
The program proofs are ideally in the order or [death program 1, death program 2, birth program 1, birth program 2]. The for loop you pointed out should skip over the death programs, so the field elements should be for the birth programs only.

Reply:
Oops, my fix had a typo in it as well. Should be fixed in my latest commit.

Reply:
okay, I just realized it skipped all the input records. (and yes, then it is reasonable)